localstack/utils/analytics/decorator.py

```diff
@@ -0,0 +1,68 @@
+import datetime
+import functools
+from multiprocessing import Process
+from typing import List
+
+import click
+
+from localstack import config
+
+from .client import AnalyticsClient
+from .events import Event, EventMetadata
+from .metadata import get_session_id
+from .publisher import AnalyticsClientPublisher
+
+ANALYTICS_API_RESPONSE_TIMEOUT_SECS = 0.5
+
+
+def _publish_cmd_as_analytics_event(command_name: str, params: List[str]):
+    event = Event(
+        name="cli_cmd",
+        payload={"cmd": command_name, "params": params},
```

Comment:
as we discussed we should probably filter these to only log the structure of the command

Reply:
not 100% sure what you mean here. In `publish_invocation` we're already filtering out the params so we only send param keys that have truthy values and not the values themselves, as we discussed.
```[k for k, v in ctx.params.items() if v]```
Are you referring to something else? Do you think that "filter" logic ought to be placed in `publish_cmd_as_analytics_event` instead?

Reply:
you're right, my mistake :-)